geode-core/src/main/java/org/apache/geode/internal/JarDeployer.java

```diff
@@ -386,6 +401,56 @@ public DeployedJar findLatestValidDeployedJarFromDisk(String unversionedJarName)
     return deployedJars;
   }
 
+  private String getJdbcDriverName(DeployedJar jar) {
+    File jarFile = jar.getFile();
+    try {
+      FileInputStream fis = new FileInputStream(jarFile.getAbsolutePath());
+      BufferedInputStream bis = new BufferedInputStream(fis);
+      ZipInputStream zis = new ZipInputStream(bis);
+      ZipEntry ze = null;
+      String driverClassName = null;
+      while ((ze = zis.getNextEntry()) != null) {
+        // JDBC 4.0 Drivers must include the file META-INF/services/java.sql.Driver. This file
+        // contains the name of the JDBC drivers implementation of java.sql.Driver
+        // See https://docs.oracle.com/javase/8/docs/api/java/sql/DriverManager.html
+        if (!ze.getName().equals("META-INF/services/java.sql.Driver")) {
+          continue;
+        }
+        int size = (int) ze.getSize();
+        byte[] b = new byte[(int) size];
+        int rb = 0;
+        int chunk = 0;
+        while (((int) size - rb) > 0) {
+          chunk = zis.read(b, rb, (int) size - rb);
+          if (chunk == -1) {
+            break;
+          }
+          rb += chunk;
+        }
+        return new String(b);
+      }
+      return null;
+    } catch (IOException ex) {
+      return null;
+    }
+  }
+
+  private static void addToSystemClasspathAndRegisterDriver(DeployedJar jar,
```

Comment:
I really don't like this workaround. Have you seen/explored using a `DriverShim` like this: http://www.kfu.com/~nsayer/Java/dyn-jdbc.html? It seems like a much cleaner solution.